server/testdriver/core/src/main/java/org/infinispan/server/test/api/HotRodTestClientDriver.java

```diff
@@ -0,0 +1,82 @@
+package org.infinispan.server.test.api;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ClientIntelligence;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.server.test.core.TestClient;
+import org.infinispan.server.test.core.TestServer;
+
+/**
+ *  REST operations for the testing framework
+ *
+ * @author Tristan Tarrant
+ * @since 10
+ */
+public class HotRodTestClientDriver extends BaseTestClientDriver<HotRodTestClientDriver> {
+   private final TestServer testServer;
+   private final TestClient testClient;
+   ConfigurationBuilder clientConfiguration;
+
+   public HotRodTestClientDriver(TestServer testServer, TestClient testClient) {
+      this.testServer = testServer;
+      this.testClient = testClient;
+
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      if(testServer.isContainerRunWithDefaultServerConfig()) {
```

Comment:
AFAICT `AbstractInfinispanServerDriver.createUserFile` runs with a non-default configuration just as well as with the default configuration, so this check isn't necessary.

Reply:
it is necessary since I only want to add this configuration for container mode and when the default file is requested (which is secured). Otherwise, users need to concern too early about security. But I don't want to make this in embedded since default config files stuff are not supported yet, and there are XML files that are not secured in our tests. So if I add a user admin, I get hard error because the server is not secured in the test but I try to connect with a user. (check the other runs errors)
I don't know how to make this better for now.

Reply:
Thanks for the explanation Katia, I thought that the users were added for every configuration.